### DISCUSSION TITLE
How to retrieve all items from a vendor?

### DISCUSSION THREAD

**Discussion by StrangeTim** [amzn/selling-partner-api-models#3794] [General] [answered]
Hi all, I'm trying to build some redundancy in my program. What I want to do is to pull all items associated with my vendor account from Amazon into a DB table. I know that it is possible to retreive individual item information using a feed or the ListingsApi but what I want to know is if there is a way to pull all items associated with a vendor account at once. The reason I want to do this is in case my DB becomes corrupted or erased. Does this functionality exist in SP-API? I'd be surprised if it doesn't but I am not seeing any of the SDKs that do what I'm trying to do.

Thanks.

**Comment by StrangeTim** [comment]
After further searching I think I've found something that will work. I think that the CatalogItems.searchCatalogItems() including the SellerID will return the list I'm looking for that I can rebuild the DB from. I'll try that and mark this as answered if it works.

**Reply by StrangeTim** [reply]
I don't think this will work after all, I have to provide identifiers or keywords for this function and all I want to do is pull all items for a seller regardless of specifications.

**Comment by chapmanjw** [comment, marked as answer]
Stay tuned for upcoming API release announcements. We are working on providing a search operation on the Listings Items API that will work very similarly to the Catalog Items API search, but will also let you paginate through listings for a vendor or seller.

**Reply by StrangeTim** [reply]
for anyone looking to do what I want before then, you can request a GET_MERCHANT_LISTINGS_ALL_DATA or similar Report (using the Reports Api) and then parse that report for the information.

**Reply by wszwc** [reply]
Hello, thank you for your suggestion. I also need to get all the product information of the seller's store, but I can't do it currently. When I use GET_MERCHANT_LISTINGS_ALL_DATA to get the report, I encountered "Creation of report schedule type 47700 is not allowed at this time." ", and I saw in the official documentation that the attributes returned by this report do not include product details. I think it cannot meet this demand. Have you found any other solutions?

**Reply by StrangeTim** [reply]
Hello wszwc, have you tried using the reports api to create an individual report of this type instead of a report schedule? Also, as far as getting the product details, you can use the SKU returned in the report and the seller Id with the Listings Items API to get all information about an item. Sure, you'll have to loop through all of the items returned in the report and get their info 1 at a time, but until the solution @chapmanjw said is in the works is done, it might be the only way to get what you need.
